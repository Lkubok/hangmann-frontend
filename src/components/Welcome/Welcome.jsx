import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
import ProblematicButton from "../ProblematicButon/ProblematicButton";
import "./Welcome.scss";

export class Welcome extends Component {
  render() {
    const { history, isLogged } = this.props;
    return (
      <div className="app-welcome">
        <div className="hang-container">
          <img className="hang-image" src="./img/hanged-man.png" alt="" />
        </div>
        <div className="welcome-buttons">
          <button
            className="button-play"
            data-tooltip="CSS EVALUATION"
            onClick={() => history.push("/game")}
          >
            {isLogged ? <>Play</> : <>Play without login for free</>}
          </button>
          {isLogged ? null : (
            <button onClick={() => history.push("/signup")}>Sign Up</button>
          )}
          <button onClick={() => history.push("/quotes")}>
            Browse available quotes
          </button>
          <ErrorBoundary>
            <ProblematicButton />
          </ErrorBoundary>
        </div>
        <div className="app-info">
          <p className="info info-header">Why play Hangman ?</p>
          <p className="info info-body">
            Learn languages with Hangman! No more dry, out of date textbook
            story lines! Here at Hangmann you'll learn languages with fun,
            interesting and cultural relevant lessons that are easy to listen
            to. But not only are they fun - they're effective too! Join the
            hundreds of thousands of people already learning languages through
            the power of this app!
          </p>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLogged: state.appParamsReducer.isLogged
});

const mapDispatchToProps = {};

Welcome.propTypes = {
  isLogged: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(Welcome);
