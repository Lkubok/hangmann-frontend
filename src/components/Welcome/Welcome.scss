@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";

.app-welcome {
  @include flex-cc-row;
  @include content-base-container;
  flex-basis: 100%;
  position: relative;
  justify-content: space-between;
  color: $primary-color;
  padding-top: 40px;
  padding-bottom: 20px;
  @media all and (max-width: 992px) {
    padding-top: 5px;
    margin-top: 0px;
    }
}
.hang-container {
  flex-basis: 40%;
  
  @media all and (max-width: 992px) {
    position: absolute;
    top: 40px;
    left: 0;
    right: 0;
    margin: 0;
    flex-basis: 100%;
    border-top: 2px solid $black-color;
  }
  

}

.hang-image {
  @media all and (max-width: 992px) {
    // width: 100%;
    }
}
.app-info {
  @include flex-cc-row;
  color: $white-color;
  font-weight: 700;
  .info {
    margin-top: 20px;
    margin-bottom: 20px;
    padding-left: 40px;
    padding-right: 40px;
    &-header {
      font-size: 2em;
      @media all and (max-width: 992px) {
      font-size: 1em;
      margin: 0;
      }
    }
    &-body {
      font-size: 1.4em;
      font-weight: 400;
      @media all and (max-width: 992px) {
        font-size: 0.7em;
        margin-bottom: 0;
        }
    }
  }
}

.welcome-buttons {
  flex-basis: 60%;
  @include flex-cc-row;
  button {
    @include welcome-button;
    flex-basis: 100%;
    margin: 20px 40px;
    padding: 20px 10px;
    @media all and (max-width: 992px) {
      margin: 5px 40px;
      padding: 20px 10px;
      }
  }
  .button-play {
    background-color: $secondary-color;
    color: $dark-color;
  }
  @media all and (max-width: 992px) {
   
    flex-basis:30%;
    position: relative;
    top: 40px;
    left: 0;
    right: 0;
    margin: 0;
    flex-basis: 100%;
  }
  button,
  .button-play {
    @media all and (max-width: 992px) {
      background-color: $dark-color-light-opacity;
      opacity: 50%;
    }
  }
  .button-play{
    background-color: $secondary-color-opacity;
  }
}

