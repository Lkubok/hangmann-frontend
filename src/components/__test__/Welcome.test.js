import React, { Component } from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Welcome } from "../Welcome/Welcome";
Enzyme.configure({ adapter: new Adapter() });

describe("Welcome component", () => {
  beforeEach(() => {});
  it("Should match snaptshot", () => {
    const wrapper = shallow(<Welcome />);
    expect(wrapper).toMatchSnapshot();
  });
});
