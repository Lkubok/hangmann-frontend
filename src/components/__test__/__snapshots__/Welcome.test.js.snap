// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component Should match snaptshot 1`] = `
<div
  className="app-welcome"
>
  <div
    className="hang-container"
  >
    <img
      alt=""
      className="hang-image"
      src="./img/hanged-man.png"
    />
  </div>
  <div
    className="welcome-buttons"
  >
    <button
      className="button-play"
      onClick={[Function]}
    >
      Play without login for free
    </button>
    <button
      onClick={[Function]}
    >
      Sign Up
    </button>
    <button
      onClick={[Function]}
    >
      Browse available quotes
    </button>
  </div>
  <div
    className="app-info"
  >
    <p
      className="info info-header"
    >
      Why play Hangman ?
    </p>
    <p
      className="info info-body"
    >
      Learn languages with Hangman! No more dry, out of date textbook story lines! Here at Hangmann you'll learn languages with fun, interesting and cultural relevant lessons that are easy to listen to. But not only are they fun - they're effective too! Join the hundreds of thousands of people already learning languages through the power of this app!
    </p>
  </div>
</div>
`;
